src/interpreters/Neorg_original.rs

```diff
@@ -0,0 +1,329 @@
+        {
+            let next_line = real_nvim_instance
+                .get_current_buf()
+                .unwrap()
+                .get_lines(&mut real_nvim_instance, line_n, line_n + 1, false)
+                .unwrap();
+            self.data.current_line = next_line.join("");
+            line_n += 1;
+        }
+
+        if self
+            .data
+            .current_line
+            .trim_start().to_lowercase()
+            .starts_with("@code")
+                {
+            let flavor = self
+                .data
+                .current_line
+                .trim_start()
+                .split_whitespace()
+                .nth(1)
+                .unwrap_or("")
+                .to_owned();
+            let end_line = real_nvim_instance
+                .get_current_buf()
+                .unwrap()
+                .line_count(&mut real_nvim_instance)
+                .unwrap();
+            let capped_end_line = std::cmp::min(line_n + 400, end_line); // in case there is a very long file, don't search for nothing up to line 500k
+            let it = line_n + 1..capped_end_line + 1;
+
+            let mut code_bloc = vec![];
+            for i in it {
+                let line_i = real_nvim_instance
+                    .get_current_buf()
+                    .unwrap()
+                    .get_lines(&mut real_nvim_instance, i - 1, i, false)
+                    .unwrap()
+                    .join("");
+                if line_i.trim_start().to_lowercase().starts_with("@end")
+                {
+                    //found end of bloc
+                    self.data.current_bloc = code_bloc.join("\n");
+                    info!(
+                        "line to extract filetype from: {:?}",
+                        line_i.split_whitespace().collect::<Vec<_>>()
+                    );
+                    return Ok(self.filetype_from_str(&flavor));
+                } else {
+                    info!("adding line {} to current bloc", i);
+                    code_bloc.push(line_i.to_string());
+                }
+            }
+        }
+
+        // if we are in a block
+        for i in (1..line_n).rev() {
+            {
+                let line_i = real_nvim_instance
+                    .get_current_buf()
+                    .unwrap()
+                    .get_lines(&mut real_nvim_instance, i - 1, i, false)
+                    .unwrap()
+                    .join("");
+                if line_i.trim_start().to_lowercase().starts_with("@code")
+                {
+                    let flavor = line_i
+                        .trim_start()
+                        .split_whitespace()
+                        .nth(1)
+                        .unwrap_or("")
+                        .to_owned();
+                    return Ok(self.filetype_from_str(&flavor));
+                }
+            }
+        }
+        Ok(String::new())
+    }
+
+    /// Convert neorg code block flavor to filetype
```

Comment:
this list appears copied from orgmode. Fine, but if the neorg standard specifies what is accepted as language hint for code bloc, this list should be replaced/updated